src/data/polynomial/eval.lean

```diff
@@ -494,8 +494,13 @@ by { rw [map, eval₂_mul_noncomm], exact λ k, (commute_X _).symm }
 @[simp] lemma map_smul (r : R) : (r • p).map f = f r • p.map f :=
 by rw [map, eval₂_smul, ring_hom.comp_apply, C_mul']
 
-/-- `polynomial.map` as a `ring_hom` -/
--- TODO: can't we make this the main definition of `polynomial.map`?
+/-- `polynomial.map` as a `ring_hom`. -/
+-- `map` is a ring-hom unconditionally, and theoretically the definition could be replaced,
+-- but this turns out not to be easy because `p.map f` does not resolve to `polynomial.map`
+-- if `map` is a `ring_hom` instead of a plain function; the elaborator does not try to coerce
```

Comment:
If it were called `ring_hom.map_polynomial` then dot notation would work fine (as `f.map_polynomial p`, but I guess that's more verbose.

Reply:
I guess `map_poly` could work? It'd be nice if we didn't have to work around this, though, because I feel like this will keep coming up; there are many maps that are probably worth bundling

Reply:
(for example `leval`, `eval_ring_hom`...)

Reply:
I don't like the idea of `ring_hom.map_poly` because I think that we should use `ring_hom_class` for `f`, not `ring_hom`. Then we'll be able to use `ring_equiv`, `alg_hom` etc without explicit casts.

Reply:
I wonder whether we could use ring_hom_class to make a specific type that works for this, maybe `(f : ring hom class) (p : polynomial R) : ring_hom_target (polynomial S)` with `ring_hom_class K -> ring_hom_target S`, although I'm not sure how you'd make that structure

Reply:
@eric-wieser @urkud are you okay with merging this doc change for now?